### PR TITLE
updates to opencover module to make it more user friendly and works on coreclr

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
added type to returned objects
replaced expand-ziparchive with call to Archive module
added defaults to location of opencover.xml, opencover installation, powershell.exe, and location of tests
updated error messages with corrective action
replaced System.Net.WebClient call (not available on coreclr) with invoke-webrequest
scoped change of executionpolicy to process when starting pester to avoid error if system policy is less restrictive
updated version of module

### CODE REVIEW

**Comment by SteveL-MSFT** [comment]
cc @adityapatwardhan

File: test/tools/OpenCover/OpenCover.psm1 [hunk 1]
```diff
@@ -1,5 +1,7 @@
 ﻿#region privateFunctions
 
+$script:psRepoPath = git rev-parse --show-toplevel
```
Comment (adityapatwardhan) [inline test/tools/OpenCover/OpenCover.psm1]: >git rev-parse --show-toplevel [](start = 21, length = 29)

git tools are not available in Azure Functions or other automation pipelines. Please check if git tools exist before using them.

Comment (SteveL-MSFT) [inline test/tools/OpenCover/OpenCover.psm1]: adding check for git

File: test/tools/OpenCover/OpenCover.psm1 [hunk 2]
```diff
@@ -359,22 +332,15 @@ function Invoke-OpenCover
     [CmdletBinding(SupportsShouldProcess=$true)]
     param ( 
         [parameter()]$OutputLog = "$pwd/OpenCover.xml",
-        [parameter(Mandatory=$true)]$TestDirectory,
-        [parameter()]$OpenCoverPath,
-        [parameter(Mandatory=$true)]$PowerShellExeDirectory,
+        [parameter()]$TestDirectory = "$($script:psRepoPath)/test/powershell",
+        [parameter()]$OpenCoverPath = "~/OpenCover",
+        [parameter()]$PowerShellExeDirectory = "$($script:psRepoPath)/src/powershell-win-core/bin/debug/netcoreapp1.0/win10-x64",
```
Comment (adityapatwardhan) [inline test/tools/OpenCover/OpenCover.psm1]: >"$($script:psRepoPath)/src/powershell-win-core/bin/debug/netcoreapp1.0/win10-x64", [](start = 47, length = 82)

This should be "$($script:psRepoPath)/src/powershell-win-core/bin/CodeCoverage/netcoreapp1.0/win10-x64"

File: test/tools/OpenCover/OpenCover.psm1 [hunk 3]
```diff
@@ -359,22 +332,15 @@ function Invoke-OpenCover
...
         [switch]$CIOnly
         )
 
     # check to be sure that OpenCover is present
 
-    if(-not $PSBoundParameters.ContainsKey('OpenCoverPath'))
-    {        
-        $openCoverBin = (Get-Command -Name 'opencover.console' -ErrorAction Ignore).Source
```
Comment (SteveL-MSFT) [inline test/tools/OpenCover/OpenCover.psm1]: I'll add a different check if opencover.console is in the path

File: test/tools/OpenCover/OpenCover.psm1 [hunk 4]
```diff
@@ -332,18 +304,19 @@ function Install-OpenCover
...
-    if ( ! (test-path $env:TEMP/$Filename) )
+    Invoke-WebRequest -Uri $packageUrl -OutFile "$tempPath"
+    if ( ! (test-path $tempPath) )
     {
         throw "Download failed: $packageUrl"
     }
     
     import-module Microsoft.PowerShell.Archive
-    Expand-ZipArchive -Path "$env:TEMP/$filename" -DestinationPath "$TargetDirectory/OpenCover"
+    Expand-Archive -Path $tempPath -DestinationPath "$TargetDirectory/OpenCover"
```
Comment (adityapatwardhan) [inline test/tools/OpenCover/OpenCover.psm1]: >Expand-Archive [](start = 4, length = 14)

This should be Expand-ZipArchive as Expand-Archive is not available on PSv4. Azure functions does not support PSv5 yet.

File: test/tools/OpenCover/OpenCover.psm1 [hunk 5]
```diff
@@ -118,43 +124,9 @@ function Get-CoverageSummary([xml.xmlelement] $element)
     }
 
     $CoverageSummary | Add-Member -MemberType ScriptMethod -Name ToString -Value { "Branch:{0,3} Sequence:{1,3}" -f $this.BranchCoverage,$this.SequenceCoverage } -Force
-        
-    return $CoverageSummary
-}
-
-function Expand-ZipArchive([string] $Path, [string] $DestinationPath)
-{
```
Comment (adityapatwardhan) [inline test/tools/OpenCover/OpenCover.psm1]: Re-instate this function as Expand-Archive is not available on PSv4.

File: test/tools/OpenCover/OpenCover.psm1 (hunk 2 above)
Comment (SteveL-MSFT) [inline test/tools/OpenCover/OpenCover.psm1]: will fix

File: test/tools/OpenCover/OpenCover.psm1 (hunk 4 above)
Comment (SteveL-MSFT) [inline test/tools/OpenCover/OpenCover.psm1]: The archive module supports ps4, can we just install it from the gallery when using Azure functions?

File: test/tools/OpenCover/OpenCover.psm1 (hunk 5 above)
Comment (SteveL-MSFT) [inline test/tools/OpenCover/OpenCover.psm1]: The archive module supports ps4, can we just install it from the gallery when using Azure functions?

Comment (adityapatwardhan) [inline test/tools/OpenCover/OpenCover.psm1]: I believe PSGet is not available on PSv4 so cannot install from gallery.

Comment (SteveL-MSFT) [inline test/tools/OpenCover/OpenCover.psm1]: PSGet supports v4, but isn't there by default.  Not a fan of duplicating expand-archive, but I guess we can keep it for now if expand-archive isn't available.

**Comment by adityapatwardhan** [comment]
@SteveL-MSFT Are you missing any format files or type files? I see you added PSTypeNames.

**Comment by SteveL-MSFT** [comment]
@adityapatwardhan in process of adding formatting, but wanted to get this merged in first.  rather than having the report as properties on the returned object, I want to have separate cmdlets for assembly coverage and the summary with independent formatting

**Comment by adityapatwardhan** [comment]
LGTM.

**Pull request merged**